### PR TITLE
refactor: vertical chapters layout for mobile screen

### PR DESCRIPTION
**Pull request by kskarthik**
This layout aligns with youtube's chapter layout on mobile

image

### CODE REVIEW

**Comment by kskarthik** [comment]
image

**Comment by FireMasterK** [comment]
On YouTube's mobile website, however, the chapters appear to be horizontal :thinking: 
image

**Comment by kskarthik** [comment]
This is how it looks to me. Looks like they are doing A/B testing?

image

**Comment by kskarthik** [comment]
image

**Comment by Bnyro** [comment]
For me it looks like for kskarthik (so vertical), weird...

**Comment by kskarthik** [comment]
@Bnyro looks like they are A/B testing the vertical/horizontal layouts. I don't know what the previous layout was. Do you know?

**Comment by Bnyro** [comment]
Nope, but does it really matter? We should choose the appearance of Piped based on how it's the most user friendly in our eyes, and not on what YouTube does.

**Comment by IronMaltese** [comment]
I prefer horizontal, but if people want vertical, it can be a toggle

**Comment by kskarthik** [comment]
should be run a poll to which layout the users prefer?

**Comment by kskarthik** [comment]
@IronMaltese  I have added feature to configure chapters layout on mobile. Patch incoming soon!

**Comment by kskarthik** [comment]
@FireMasterK done!

**Comment by kskarthik** [comment]
image

File: src/components/PreferencesPage.vue [hunk 1]
```diff
@@ -511,6 +528,10 @@ export default {
             this.disableLBRY = this.getPreferenceBoolean("disableLBRY", false);
             this.proxyLBRY = this.getPreferenceBoolean("proxyLBRY", false);
             this.hideWatched = this.getPreferenceBoolean("hideWatched", false);
+            this.mobileChapterLayout = this.getPreferenceString(
+                "mobileChapterLayout",
+                localStorage.getItem("mobileChapterLayout") || "Vertical",
+            );
```
Comment (Bnyro) [inline src/components/PreferencesPage.vue:534]:
```suggestion
            this.mobileChapterLayout = this.getPreferenceString("mobileChapterLayout", "Vertical");
```

File: src/components/PreferencesPage.vue [hunk 2]
```diff
@@ -126,6 +126,22 @@
             @change="onChange($event)"
         />
     </label>
+    <!-- chapters layout on mobile -->
+    <label class="pref" for="chkMinimizeChapters">
```
Comment (Bnyro) [inline src/components/PreferencesPage.vue:130]: I guess the pref should only be visible on mobile devices. You can just hide it using sth like `style="md:hidden"` ( not tested)

File: src/components/PreferencesPage.vue [hunk 3]
```diff
@@ -126,6 +126,22 @@
...
+            <option v-t="'video.chapters_horizontal'" value="Horizontal" />
+            <option v-t="'video.chapters_vertical'" value="Vertical" />
+        </select>
+        <!-- <input
+            id="chkMinimizeChapters"
+            v-model="minimizeChapters"
+            class="checkbox"
+            type="checkbox"
+            @change="onChange($event)"
+        /> -->
```
Comment (Bnyro) [inline src/components/PreferencesPage.vue:143]: Please remove that

**Comment by Bnyro** [comment]
Works good from my testing, just some nitpicks

**Comment by Bnyro** [comment]
Also, please don't tag team members here, it won't speed up things :)

File: src/components/PreferencesPage.vue (hunk 2 above)
Comment (kskarthik) [reply src/components/PreferencesPage.vue:130]: done

File: src/locales/en.json [hunk 4]
```diff
@@ -168,7 +169,10 @@
         "live": "{0} Live",
         "shorts": "Shorts",
         "all": "All",
-        "category": "Category"
+        "category": "Category",
+				"chapters_horizontal": "Horizontal",
+				"chapters_vertical": "Vertical"
```
[the remark this answers is not in the document]
Comment (Bnyro) [reply src/locales/en.json:174]: They still don't seem to be aligned properly, you likely need to fix the indentation manually

[the remark this answers is not in the document]
Comment (kskarthik) [reply src/locales/en.json]: Ok, I manually did with prettier, It worked!

**Comment by kskarthik** [comment]
just found that `yarn format` ignores json files, i have added support for them too

**Review by Bnyro — APPROVED** [review]
Thanks!

**Pull request merged**